### PR TITLE
nixos-rebuild: use remote $PATH in buildHostCmd

### PR DESCRIPTION
**Pull request by fricklerhandwerk**
## Motivation for this change

I was trying to port `nixos-rebuild --target-host <target>` [to Darwin](https://discourse.nixos.org/t/how-to-run-nixos-rebuild-target-host-from-darwin/9488/3?u=fricklerhandwerk) and noticed that the `$PATH` to run commands on the build host was actually the local one. It should of course be the remote one.

This fixes the issue when using

    nixos-rebuild --target-host <target> --use-remote-sudo switch

and the local machine does not have anything in `$PATH` that would resolve to `sudo` on the remote machine.

## Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- [x] Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date (no change)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by cole-h** [comment]
@ofborg eval

**Comment by fricklerhandwerk** [comment]
@zimbatm @lopsided98 @grahamc You've been working on that code in the past ~1 year, can anyone take a quick look?

**Comment by zimbatm** [comment]
Good catch, but I don't think that the solution is working as expected. Did you test this?
```console
$ BAR=3 env 'FOO=$BAR' bash -c 'echo $FOO'
$BAR
```

**Comment by cole-h** [comment]
I think the point is that `$PATH` won't be expanded locally (e.g. on the system executing the `ssh`), but on the remote system, which is why `$PATH` is surrounded in single-quotes -- to prevent bash from expanding it to the local host's `$PATH`.

e.g. the `PATH` will look like `/nix/store/......../bin:$PATH` and not e.g. `/nix/store/........./bin:/run/wrappers/bin:/home/vin/.nix-profile/bin:/etc/profiles/per-user/vin/bin:/nix/var/nix/profiles/default/bin:/run/current-system/sw/bin` (from the host system).

TL;DR: It's expected the single quotes prevent expansion on the local system. Compare `ssh <somehost> echo $PATH` with `ssh <somehost> echo '$PATH'`.

**Comment by fricklerhandwerk** [comment]
@cole-h Thanks for the explanation, I have amended the commit message to make it clearer for the future reader.

**Comment by zimbatm** [comment]
I don't understand what does the path expansion, but it seems to work.

```console
$ ssh build01.nix-community.org env FOO='$HOSTNAME' env | grep FOO
FOO=nix-community-build01
```

**Review by zimbatm — APPROVED** [review]

**Review by cole-h — APPROVED** [review]
Thanks!
